### ISSUE TITLE
Check that there is no space in cluster for put object when we put object and in-memory cache is on

### ISSUE THREAD

**Issue by alexchetaev** [nspcc-dev/neofs-node#1703]
The issue is that currently we don't check that there is a free disk space available in cluster for put object when we write it through in memory cache. So It can easily seem to the situation then we need to add more disk but can do it. If we restart service for add more disk all object in memory will be lost.

We need the mode for storage node when we check that we have space on the disk before write in memory and return successful response to client.

**Comment by cthulhu-rider** [comment]
Such a pre-check won't fix the issue completely: there could be enough storage space on the check moment, and not enough - micro-interval after. I'd reconsider the exec approach with some sort of `sync` flag in the request which preserves data to be left in memory.

**Comment by fyrchik** [comment]
After #1584 , we no longer have in-memory cache, so the issue not applies.
